src/cryptsetup/cryptsetup-tpm2.c

```diff
@@ -196,6 +304,31 @@ int find_tpm2_auto_data(
                         return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
                                                "Invalid base64 data in 'tpm2-policy-hash' field.");
 
+                /* The flags field is optional and used to encode newly introduced features */
+                w = json_variant_by_key(v, "tpm2-flags");
+                if (w) {
+                        if (!json_variant_is_array(w)) {
+                                return log_error_errno(
+                                        SYNTHETIC_ERRNO(EINVAL),
+                                        "TPM2 token data flags field is not an array.");
+                        }
+                        JSON_VARIANT_ARRAY_FOREACH(e, w) {
+                                const char *fs = NULL;
+                                if (!json_variant_is_string(e))
+                                        return log_error_errno(
+                                                SYNTHETIC_ERRNO(EINVAL),
+                                                "TPM2 token data flag is not a string.");
+                                fs = json_variant_string(e);
+                                if (!fs)
```

Comment:
given you already check that it's a string just a few lines up this cannot fail. hence use assert() or so for this.